### DISCUSSION TITLE
WIFI Restarting if DNSCrypt DNS is used in WIndows 11

### DISCUSSION THREAD

**Discussion by poqdavid** [DNSCrypt/dnscrypt-proxy#1983] [Q&A] [answered]
Every time I add DNSCrypt to my WIFI it disconnects itself from AP and then reconnects once no idea why

Here is my DNSCrypt config https://gist.github.com/poqdavid/b0894d80bdf68114f1e04e40c2ac5a6a

And I get this in my event logs
```
WLAN AutoConfig detected limited connectivity, attempting automatic recovery.
```

Does anyone know anything about this and how I can fix it?

Update: Noticed when Network Location Awareness service is disabled AP doesn't always disconnect but sometimes

Why Network Location Awareness and DNSCrypt mix is causing this is this a problem with Windows or DNSCrypt?

OS: Windows 11 Pro 64bit
WIFI: Intel(R) Wireless-AC 9560 160MHz (latest driver)

**Comment by lessload** [comment]
Same problem here. No way to fix it. Microsoft NCSI did it. Looks like they don't want us to use dnscrypt-proxy.

**Comment by lifenjoiner** [comment, marked as answer]
> Every time I add DNSCrypt to my WIFI

Every time? Is this required often? What is the scenario? Once you have set a static DNS, won't it be saved?

> it disconnects itself from AP and then reconnects once

Once the config of the network is been changed, the OS redoes the initialization.

**Reply by poqdavid** [reply]
> Every time? Is this required often? What is the scenario? Once you have set a static DNS, won't it be saved?
> Once the config of the network is been changed, the OS redoes the initialization.

No by that I meant that I tested it with normal DNS servers and the Wi-Fi was all fine and once I added the DNSCrypt again it happened again
Like if DNSCrypt is used `Network Location Awareness` service crashes the Wi-Fi once and then auto connects just fine

**Reply by lifenjoiner** [reply]
Maybe you can try more referring to https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Installation-Windows#troubleshooting

**Reply by poqdavid** [reply]
Looks like it's working but I remember I had that enabled before hmm something must have disabled it
Also just enabled `Captive portal` which I didn't enable before
and going to see it for a few days hope it stays fixed

**Reply by lifenjoiner** [reply]
Maybe the updates reset some settings. Win 11 is buggier ;)

**Reply by poqdavid** [reply]
True I also added the DoH to the list of known servers on Windows 11 It all works just fine too

**Comment by issuex** [comment]
Oh.! found the Same problem, Same Event log. I need to reconnect wifi twice, every time i need to connect the wifi.
